### PR TITLE
Plans: don't hide on small screens

### PR DESCRIPTION
**Pull request by mattwiebe**
#### Changes proposed in this Pull Request:

* don't hide the plans table on small screens

#### Testing instructions:

View the Plans tab on a screen smaller than 600px wide. Before this PR, the table will be hidden. After, it shows.

It does look bad on very small screens, and we should plan to bring over the plans cards view from Calypso in the next release.

### CODE REVIEW

**Comment by jeffgolenski** [comment]
@mattwiebe + @oskosk let me test this a bit and we'll see what kind of a quick fix we can get in until we can get the full collapsible plans cards from calypso (and jetpack.com). cc @keoshi

**Comment by jeffgolenski** [comment]
Alright, I made some quick enhancements for a slightly better UX on small mobile devices

Currently, on screen sizes smaller than 660px, we can see some real cramping in the table:

screen shot 2018-03-07 at 3 01 28 pm

Unfortunately, after taking a look at the code, I discovered that this table is built with some row-centric table markup (instead of column centric), so we're not easily able to break apart the columns and make them mobile friendly.

Because of which, I'm suggesting we hide the table on devices with screen sizes smaller than 660px and display a card with two CTAs. One with a link to manage the users plan, and one to view all plans. I've already coded them in and it currently looks like this on small screens:

screen shot 2018-03-07 at 2 52 38 pm

These links should take the users to view the proper plans page and plans listing in calypso, which are all mobile optimized.

@mattwiebe I just need a little help from you!
[ ] Take a look at my code in the plans-grid.jsx file. It probably could use some cleaning up to match proper react syntax
[ ] Make the "manage your plan" link take the user to `https://wordpress.com/plans/my-plan/sitename.com
[ ] Make the "view all plans" link take the user to `https://wordpress.com/plans/sitename.com`

Thankfully, we have a really low traffic rate for small mobile devices at this point in time, so this is an okay quick fix, but we'll need to take some time to look at bringing over the proper pricing tables from calypso at some point since they're all optimized for mobile!

Thanks!

File: _inc/client/plans/plan-grid.jsx [hunk 1]
```diff
@@ -41,6 +41,11 @@ class PlanGrid extends React.Component {
 
 		return (
 			<div className="plan-features">
+				<div className="plans-mobile-notice dops-card">
+					<h2>Your plan</h2>
+					<p>You're currently on the Jetpack [plan name].</p>
```
Comment (zinigor) [inline _inc/client/plans/plan-grid.jsx]: This line actually needs the plan name now, and the entire block needs translated strings.

**Comment by designsimply** [comment]
For reference, here's what I see for WP 4.9.4 JP 5.9 for WP Admin > Jetpack > "Plans" and "Compare All Plans."

   

And here is what I see on the same page with the branch loaded via https://jurassic.ninja/create/?shortlived&jetpack-beta&nojetpack&branch=fix/plans-small-screens 

1. Note the placeholder text on for "[plan name]". 
1. When I tapped "View all Jetpack plans" I was prompted to select a site (even though I just came from a very specific site), is that expected?
1. Looks like my test site isn't loading the new layout from the branch and I suspect a Pixel phone doesn't meet the smaller than 600px requirement.

  

<sup>Tested with Pixel, Android 8.1.0 using WordPress 4.9.4 and Jetpack 6.0-beta-17828-4f10116-fix_plans-small-screens.</sup>

**Comment by jeffgolenski** [comment]
@zinigor @designsimply Thanks for testing. Sorry, this PR should of had a "in progress" label—there's still work to be done here!

**Comment by jeffgolenski** [comment]
@mattwiebe Lookin' good dude.

**Pull request merged**